pypesto/visualize/observable_mapping.py

```diff
@@ -37,27 +41,352 @@
     pass
 
 
+def visualize_estimated_observable_mapping(
+    pypesto_result: Result,
+    pypesto_problem: HierarchicalProblem,
+    start_index: int = 0,
+    axes: Optional[plt.Axes] = None,
+    **kwargs,
+):
+    """Visualize the estimated observable mapping for relative and semi-quantitative observables.
+
+    Visualizes the estimated linear mapping for relative observables and the non-linear
+    spline approximation for semi-quantitative observables.
+
+    Parameters
+    ----------
+    pypesto_result:
+        The pyPESTO result object from optimization.
+    pypesto_problem:
+        The pyPESTO problem.
```

Comment:
The pypesto problem objective is used downstream and I think that we can have problems without objective so it could be nice to specify here that the `pypesto_problem` is expected to have an objective

Reply:
True.
Added it into the docstring and added some check to check that there is an objective and that it has the correct calculator.